src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -291,19 +291,27 @@ protected function doGenerate(array $variables, array $defaults, array $requirem
         }
 
         // add a query string if needed
-        $extra = array_udiff_assoc(array_diff_key($parameters, $variables), $defaults, function ($a, $b) {
+        $extras = array_udiff_assoc(array_diff_key($parameters, $variables), $defaults, function ($a, $b) {
             return $a == $b ? 0 : 1;
         });
 
+        foreach ($extras as $param => $value) {
+            if (null === $value || \is_array($value)) {
+                continue;
+            }
+
+            $extras[$param] = (string) $value;
+        }
+
         // extract fragment
         $fragment = $defaults['_fragment'] ?? '';
 
-        if (isset($extra['_fragment'])) {
-            $fragment = $extra['_fragment'];
-            unset($extra['_fragment']);
+        if (isset($extras['_fragment'])) {
+            $fragment = $extras['_fragment'];
+            unset($extras['_fragment']);
         }
 
-        if ($extra && $query = http_build_query($extra, '', '&', \PHP_QUERY_RFC3986)) {
+        if ($extras && $query = http_build_query($extras, '', '&', \PHP_QUERY_RFC3986)) {
```

Comment:
http_build_query doesn't cast? I didn't know that!

Reply:
Me too, but it looks like this is the problem here

Reply:
could be an interesting thing for PHPStan

cc @ondrejmirtes

Reply:
Hi, please open a feature request in PHPStan repo with description 😊

Reply:
Here you go: https://github.com/phpstan/phpstan/issues/5286